plugin/forward/persistent.go

```diff
@@ -159,8 +159,8 @@ func (t *Transport) SetExpire(expire time.Duration) { t.expire = expire }
 func (t *Transport) SetTLSConfig(cfg *tls.Config) { t.tlsConfig = cfg }
 
 const (
-	defaultExpire      = 10 * time.Second
-	minDialTimeout     = 100 * time.Millisecond
-	maxDialTimeout     = 30 * time.Second
-	defaultDialTimeout = 30 * time.Second
+	defaultExpire  = 10 * time.Second
+	minDialTimeout = 1 * time.Second
+	maxDialTimeout = 30 * time.Second
+	readTimeout    = 5 * time.Second // this is the normal value for some resolve take quite a while, usually (cached) responses are fast.
```

Comment:
originally (before dynamic timeout) we used 2s as timeout (https://github.com/coredns/coredns/blob/master/plugin/forward/proxy.go#L84). 5s is too much because we won't be able to retry the query via different upstreams. 
As I mentioned in #2306, I would prefer to have this timeout configurable. If not, I would prefer value 1.5s to be able to make at least 3 attempts in 5s.
Yes, resolution of some domain names may take longer time on first query, but if client will send the same request again to the same resolver - this may get resolved quickly with previously cached value